sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpPipelinePosition.java

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.clientcore.core.http.pipeline;
+
+import io.clientcore.core.utils.ExpandableEnum;
+
+/**
+ * Enum representing where in an {@link HttpPipeline} a given {@link HttpPipelinePolicy} is positioned when added to an
+ * {@link HttpPipelineBuilder}.
+ * <p>
+ * By default, policies are added at {@link #AFTER_RETRY} with the assumption of
+ * {@link HttpPipelinePolicy policies} are idempotent and may need to be updated each time a network request is sent
+ * (each try) and may mutate the results of authentication policies.
+ * <p>
+ * HttpPipelinePosition ensures that the creation of an {@link HttpPipeline} follows a strict positioning of policies.
+ * This positioning is based on what is believed to be the best position of policy execution. Here is a visual
```

Comment:
```suggestion
 * Here is a visual
```

I don't love the "believed to be" phrasing. I tried writing a couple different stabs which I also didn't love:
`This positioning is optimized for optimal policy execution.`
`The order is recommended for retrying all phases which might impact the outgoing request.`

Ultimately it just feels somewhat duplicative of the earlier comment ("are idempotent ...") to me?

Reply:
Let's just drop it for now as this is near the extreme end of configuration and likely won't be used much.